ruby/ql/lib/codeql/ruby/dataflow/internal/FlowSummaryImplSpecific.qll

```diff
@@ -257,6 +257,12 @@ ArgumentPosition parseParamBody(string s) {
   or
   s = "block" and
   result.isBlock()
+  or
+  s = "any" and
+  result.isAny()
+  or
+  s = "any-named" and
+  result.isKeyword(_)
```

Comment:
We need a new `TAnyKeywordArgumentPosition` for this instead.

Reply:
I can certainly add that :+1: Can you elaborate on why?

The current solution certainly does cause number of results for `parseParamBody` to grow needlessly, since `"any-named"` will have results for all keyword arguments. However, that is 2x at most, if all arguments are positional, which I wouldn't assume to have a major impact on performance (although it certainly doesn't help).

I guess I'm just curious whether this is to ensure we have a clean codebase, or whether I should maybe have worried more about the performance aspect.

Reply:
It is for performance reasons: Otherwise a MaD row containing `Argument[any-named]` will expand to a row for each possible named argument, which we don't want. Instead, we push that fan-out into the inlined predicate `parameterMatch`, and let the data-flow library make sure to only apply the fan-out when needed.